examples/delegated-fungible/src/lib.rs

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) Zefchain Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+/* ABI of the Delegated Fungible Token Example Application */
+
+use async_graphql::{scalar, Request, Response};
+pub use linera_sdk::{
+    abi::{ContractAbi, ServiceAbi},
+    abis::fungible::{Account, InitialState, Parameters},
+    graphql::GraphQLMutationRoot,
+    linera_base_types::{AccountOwner, Amount},
+};
+use serde::{Deserialize, Serialize};
+
+/// An ABI for applications that implement a fungible token.
+pub struct DelegatedFungibleTokenAbi;
+
+impl ContractAbi for DelegatedFungibleTokenAbi {
+    type Operation = Operation;
+    type Response = ();
+}
+
+impl ServiceAbi for DelegatedFungibleTokenAbi {
+    type Query = Request;
+    type QueryResponse = Response;
+}
+
+/// A delegated fungible operation
+#[derive(Debug, Deserialize, Serialize, GraphQLMutationRoot)]
+pub enum Operation {
+    /// Approve the transfer of tokens
+    Approve {
+        /// Owner to transfer from
+        owner: AccountOwner,
+        /// The spender account
+        spender: AccountOwner,
+        /// Maximum amount to be transferred
+        allowance: Amount,
+    },
+    /// Transfers tokens from a (locally owned) account to a (possibly remote) account.
+    Transfer {
+        /// Owner to transfer from
+        owner: AccountOwner,
+        /// Amount to be transferred
+        amount: Amount,
+        /// Target account to transfer the amount to
+        target_account: Account,
+    },
+    /// Transfers tokens from a (locally owned) account to a (possibly remote) account by using the allowance.
+    TransferFrom {
```

Comment:
Original ([from ERC20](https://docs.openzeppelin.com/contracts/4.x/api/token/erc20#IERC20-transferFrom-address-address-uint256-)) `transferFrom` has 3 arguments:
- from
- to
- amount

Why does our version need to have 4?

Reply:
In the ERC20, they use `msg.sender`, which is then the spender.